### ISSUE TITLE
Change to protocol A makes no improvment

### ISSUE THREAD

**Issue by bratao** [LINBIT/linstor-server#119]
Hello, 
I´m writing an article comparing Linstor to other solutions for k8s like StorageOs and OpenEBS. The read performance is exactly if I was benchmarking the disk directly, but the writing performance is very low. I´m trying to tweak the options to an async protocol, but the benchmark had no changes.
These are 4 machines on the same datacenter, with latency of 0.2s. All have a fast NVMe disk.

But setting to the A Protocol and a larger buffer apparently makes no difference:

With:
linstor controller drbd-options --protocol A --sndbuf-size 10485760 --rcvbuf-size 10485760 --max-buffers 131072 --disk-flushes no --c-fill-target 24000

```
==================
= Dbench Summary =
==================

Random Read/Write IOPS: 294k/22.3k. BW: 9601MiB/s / 111MiB/s
Average Latency (usec) Read/Write: 13.83/1289.22
Sequential Read/Write: 8869MiB/s / 113MiB/s
Mixed Random Read/Write IOPS: 66.2k/22.3k
```

With:
linstor controller drbd-options --protocol C --sndbuf-size 0 --rcvbuf-size 0 --max-buffers 2000 --disk-flushes yes --c-fill-target 100

```
==================
= Dbench Summary =
==================
Random Read/Write IOPS: 322k/22.5k. BW: 9509MiB/s / 112MiB/s
Average Latency (usec) Read/Write: 14.62/1295.31
Sequential Read/Write: 8823MiB/s / 112MiB/s
Mixed Random Read/Write IOPS: 66.8k/22.2k

```

I need to change it in another place? Am I doing something wrong?

**Comment by ghernadi** [comment]
Hello,

as you have closed this issue, I assume you have figured out the reason? Would you mind briefly explain your findings here for future readers?

**Comment by bratao** [comment]
@ghernadi I did more tests and debuging and I realized that Protocol A is good only until the TCP buffer fill-up. The benchmark I was running ( https://github.com/leeliu/dbench) it's too demanding for the TCP buffer do any difference and I was using the O_DIRECT flag.
With the regular Linux cache I got satisfactory write speed and the eventual data loss was acceptable to my use case.
Correct me if I'm wrong but the answer for reability and speed is drbd-proxy. That is a commercial product, that I cannot afford but looks like an fantastic and worth tool.
